### ISSUE TITLE
"Error: Error getting data key: 0 successful groups required, got 0" /w GCP

### ISSUE THREAD

**Issue by byakku** [getsops/sops#934]
Hello!

We're using sops to work with files encrypted via GCP KMS, we also use Terraform and [terraform provider for sops](https://github.com/carlpett/terraform-provider-sops) 

While using TF we are getting weird error:

```
Error: Error getting data key: 0 successful groups required, got 0
```

[Relevant code](https://github.com/mozilla/sops/blob/master/usererrors.go#L38)

Environment:
    

- Provider: GCP, ServiceAccount with Editor Privileges 
-  Host: GKE
-  Atlantis: `v0.17.0` + Google SDK + Sops installed on container (`3.7.0`)
-   Terraform versions: `0.13.7` and `1.0.5` both tested, same error
-   sops provider: 0.6.3
-   sops version: `go.mozilla.org/sops/v3 v3.7.0`

I cannot reproduce the issue locally, unfortunately.

More details about the configuration & host in this [issue](https://github.com/carlpett/terraform-provider-sops/issues/74#issue-979189014)

I've seen people getting into this error with AWS provider but not with GCP. 

Any idea how could I debug this further?

**Comment by muhlba91** [comment]
I ran into a similar issue using ArgoCD and ksops (which is built on sops): https://github.com/viaduct-ai/kustomize-sops/issues/178

TL;DR: I can decrypt locally using the environment variable and a service account but in the ArgoCD pod it doesn't work with the exact same environment variables and service account configuration.
